src/main/kotlin/org/elm/lang/core/resolve/reference/LexicalValueReference.kt

```diff
@@ -14,10 +14,19 @@ class LexicalValueReference(element: ElmReferenceElement)
             emptyArray()
 
     override fun resolveInner(): ElmNamedElement? {
-        val resolved = getCandidates().find { it.name == element.referenceName }
+        val resolved = resolveShallow()
         return (resolved as? ElmReferenceElement)?.reference?.resolve() ?: resolved
     }
 
+    /**
+     * References of this type normally recursively resolve references to elements that are
+     * themselves references (e.g. usage of a name in a record destructuring pattern). This function
+     * will only resolve the first level of reference.
+     */
+    fun resolveShallow(): ElmNamedElement? {
```

Comment:
Is this too much of a hack? The recursive resolution is already pretty hacky.

Reply:
It's not ideal, but I think this is the world we live in. I seem to remember similar things in intellij-rust.